TestHelper.psm1

```diff
@@ -201,3 +204,209 @@ function Install-ResourceDesigner {
+
+    .EXAMPLE
+        $TestEnvironment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            -TestType Integration
+            
+        This command will initialize the test enviroment for Integration testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.    
+#>
+function Initialize-TestEnvironment
+{
+    [OutputType([PSObject])]
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCModuleName,
+        
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCResourceName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('Unit','Integration')]
+        [String] $TestType
+    )
+    
+    Write-Verbose -Verbose (`
+        'Initializing Test Environment for {0} testing of {1} in module {2}.' `
+            -f $TestType,$DSCResourceName,$DSCModuleName)   
+    if ($TestType -eq 'Unit')
+    {
+        [String] $RelativeModulePath = "DSCResources\$DSCResourceName\$DSCResourceName.psm1"
+    }
+    else
+    {
+        [String] $RelativeModulePath = "$DSCModuleName.psd1"
+    }
+
+    # Unique Temp Working Folder - always gets removed on completion
+    # The tests can put anything in here and it will get cleaned up.
+    [String] $RandomFileName = [System.IO.Path]::GetRandomFileName()
+    [String] $WorkingFolder = Join-Path -Path $env:Temp -ChildPath "$DSCResourceName_$RandomFileName" 
+    # Create the working folder if it doesn't exist (it really shouldn't anyway)
+    if (-not (Test-Path -Path $WorkingFolder))
+    {
+        New-Item -Path $WorkingFolder -ItemType Directory
+    } 
+    
+    # The folder where this module is found
+    [String] $moduleRoot = $pwd
+
+    # The folder that all tests will find this module in
+    [string] $modulesFolder = Resolve-Path -Path (Join-Path -Path $pwd -Child '..')   
+        
+    # Import the Module
+    $Splat = @{
+        Path = $moduleRoot
+        ChildPath = $RelativeModulePath
+        Resolve = $true
+        ErrorAction = 'Stop'
+    }
+    $DSCModuleFile = Get-Item -Path (Join-Path @Splat)
+    
+    # Remove all copies of the module from memory so an old one is not used.
+    if (Get-Module -Name $DSCModuleFile.BaseName -All)
+    {
+        Get-Module -Name $DSCModuleFile.BaseName -All | Remove-Module
+    }
+    
+    # Import the Module to test.
+    Import-Module -Name $DSCModuleFile.FullName -Force
+    
+    # Set the PSModulePath environment variable so that the module path this module is in
+    # appears first because the LCM will use this path to try and locate modules when integration
+    # tests are called. This is to ensure the correct module is tested.
+    [String] $OldModulePath = $env:PSModulePath
+    [String] $NewModulePath = $OldModulePath
+    if (($NewModulePath).Split(';') -ccontains $modulesFolder)
```

Comment:
I think the way he wrote it is better, specifically it also covers the event that the folder name is contained in the ModulePath but is not the first one in particular. As it's possible the machine has the folder the user is working out of but is the first one.  This way we don't append more folders on then needed.